components/scream/tests/uncoupled/homme/homme_standalone.cpp

```diff
@@ -69,6 +73,29 @@ TEST_CASE("scream_homme_standalone", "scream_homme_standalone") {
   // Init, run, and finalize
   ad.initialize(atm_comm,ad_params,t0);
 
+  // Check that topography data from the FM matches Homme.
+  {
+    auto& geo = Homme::Context::singleton().get<Homme::ElementsGeometry>();
+    auto phis = geo.m_phis;
+
+    const auto& atm_process_group = ad.get_atm_processes();
+    const auto& process = atm_process_group->get_process(0);
+    const scream::HommeDynamics* homme_process = static_cast<const scream::HommeDynamics*>(&(*process));
+    const auto phinh_i = homme_process->get_internal_field("phinh_i","Dynamics").get_view<Real*****>();
+
+    int nelem = Homme::Context::singleton().get<Homme::ElementsGeometry>().num_elems();
+    int n0 = Homme::Context::singleton().get<Homme::TimeLevel>().n0;
+    constexpr int NVL = HOMMEXX_NUM_PHYSICAL_LEV;
+
+    Kokkos::parallel_for(Kokkos::RangePolicy<>(0,nelem*NP*NP),
+                         KOKKOS_LAMBDA (const int idx) {
+      const int ie = idx/(NP*NP);
+      const int ip = (idx/NP)%NP;
+      const int jp = idx%NP;
+      EKAT_KERNEL_ASSERT(phinh_i(ie,n0,ip,jp,NVL) == phis(ie,ip,jp));
+    });
+  }
```

Comment:
I add a check that the internal dyn field `phinh_i(nlevs)` matches internal homme `phis`.